apps/ewallet_db/test/ewallet_db/transaction_consumption_test.exs

```diff
@@ -0,0 +1,175 @@
+defmodule EWalletDB.TransactionConsumptionTest do
+  use EWalletDB.SchemaCase
+  alias EWalletDB.TransactionConsumption
+
+  describe "TransactionConsumption factory" do
+    test_has_valid_factory TransactionConsumption
+  end
+
+  describe "get/1" do
+    test "returns an existing transaction consumption" do
+      inserted = insert(:transaction_consumption)
+      consumption = TransactionConsumption.get(inserted.id)
+      assert consumption.id == inserted.id
+    end
+
+    test "returns nil if the transaction consumption does not exist" do
+      consumption = TransactionConsumption.get("unknown")
+      assert consumption == nil
+    end
+  end
+
+  describe "get/2" do
+    test "returns nil if the transaction consumption does not exist" do
+      consumption = TransactionConsumption.get("unknown")
+      assert consumption == nil
+    end
+
+    test "preloads the specified association" do
+      inserted = insert(:transaction_consumption)
+      consumption = TransactionConsumption.get(inserted.id, preload: [:minted_token])
+      assert consumption.id == inserted.id
+      assert consumption.minted_token != nil
+    end
+  end
+
+  describe "expire_all/0" do
+    test "expires all requests past their expiration date" do
+      now = NaiveDateTime.utc_now()
+
+      # t1 and t2 have expiration dates in the past
+      t1 = insert(:transaction_consumption, expiration_date: NaiveDateTime.add(now, -60, :seconds))
+      t2 = insert(:transaction_consumption, expiration_date: NaiveDateTime.add(now, -600, :seconds))
+      t3 = insert(:transaction_consumption, expiration_date: NaiveDateTime.add(now, 600, :seconds))
+      t4 = insert(:transaction_consumption, expiration_date: NaiveDateTime.add(now, 160, :seconds))
+
+      # They are still valid since we haven't made them expired yet
+      assert TransactionConsumption.expired?(t1) == false
+      assert TransactionConsumption.expired?(t2) == false
+      assert TransactionConsumption.expired?(t3) == false
+      assert TransactionConsumption.expired?(t4) == false
+
+      TransactionConsumption.expire_all()
+
+      # Reload all the records
+      t1 = TransactionConsumption.get(t1.id)
+      t2 = TransactionConsumption.get(t2.id)
+      t3 = TransactionConsumption.get(t3.id)
+      t4 = TransactionConsumption.get(t4.id)
+
+      # Now t1 and t2 are expired
+      assert TransactionConsumption.expired?(t1) == true
+      assert TransactionConsumption.expired?(t2) == true
+      assert TransactionConsumption.expired?(t3) == false
+      assert TransactionConsumption.expired?(t4) == false
+    end
+
+    test "sets the expired_at field" do
+      now = NaiveDateTime.utc_now()
+      t = insert(:transaction_consumption, expiration_date: NaiveDateTime.add(now, -60, :seconds))
+      TransactionConsumption.expire_all()
+      t = TransactionConsumption.get(t.id)
+
+      assert TransactionConsumption.expired?(t) == true
```

Comment:
**Request:** Can you also assert that `expired?(t) == false` and `expired_at == nil` before `expire_all()`?